trek.js

```diff
@@ -0,0 +1,149 @@
+      response.data.forEach((trip) => {
+        tripList.append(`<li>${trip.name}</li>`)
+
+        // within the first GET request response, nested function that shows details of a specific trip
+        const showDetails = () => {
+
+          const tripDetail = () => {
+            axios.get(TRIP_URL + trip.id)
+            .then((response) => {
+              const detail = $('.details')
+              detail.empty();
+
+              // populating HTML with details from the second GET request response
+              detail.html(`<h1>Is This the Trip of Your Dreams?<h1> 
+                <h2>Excursion Name: <span>${response.data.name}</span></h2>
+                <h3>Continent: <span>${response.data.continent}</span></h3>
+                <h3>Excursion Category: <span>${response.data.category}</span></h3>
+                <h3>Weeks of Travel: <span>${response.data.weeks}</span></h3>
+                <h3>Cost: <span>$${response.data.cost}</span></h3>
+                <h3>Why This Trek Is Amazing: </h3>
+                <p>${response.data.about}</p>`)
+
+                // displays the form only when a specific trip is clicked
+                const form = $('#trek-form')
+                form.html(`<h1>Book Your Trip Now!</h1>
+                          <form id="trek-form">
+                          <div>
+                            <label for="name">Name</label>
+                            <input type="text" name="name" />
+                          </div>
+        
+                          <div>
+                            <label for="age">Age</label>
+                            <input type="number" name="age" />
+                          </div>
+  
+                          <div>
+                              <label for="email">Email</label>
+                              <input type="text" name="email" />
+                            </div>
+        
+                          <input type="submit" name="add-trek" value="Book My Trek" />
+                        </form>`)
+
+                  // function to perform the addTrek function on a click event.
+                  const addTrek = (event) => {
+
+                    event.preventDefault();
+                        
+                    const trekData = readFormData();
+                        
+                    reportStatus("About to book your trip!");
+                    // uses the trip.id from our previous response     
+                    axios.post(`${TRIP_URL}${trip.id}/reservations`, trekData)
+                    .then((response) => {
+                       
+                      const trekID = response.data.id;
+                      reportStatus(`Successfully booked a new trip with ID ${trekID}`);
+                      clearForm();
+                    })
+                    .catch((error) => {
+                      reportApiError(error);
+                    })
+                    };
+
+                    $('#trek-form').submit(addTrek);
```

Comment:
Look where you have this.  Every time I click on a trip to see the details this line adds a **new** event listener.  So if I click on 3 trips to see the details and then fill out the form to submit, **three** submit handlers will fire.  So the app will make 3 post requests.